CMakeLists.txt

```diff
@@ -63,51 +62,102 @@ set( TRACE_SOURCE_FILES
 
 add_library( ${LIBRARY_NAME} ${TRACE_SOURCE_FILES} )
 
+include( GNUInstallDirs )
+
 set_target_properties( ${LIBRARY_NAME} PROPERTIES
     CXX_STANDARD          17
     CXX_STANDARD_REQUIRED ON
     CXX_EXTENSIONS        OFF
     COMPILE_FLAGS         "${TRACE_COMPILE_FLAGS} ${TRACE_FLAGS_COVERAGE}"
-    LINK_FLAGS            "${TRACE_LINK_FLAGS}    ${TRACE_FLAGS_COVERAGE}" )
+    LINK_FLAGS            "${TRACE_LINK_FLAGS}    ${TRACE_FLAGS_COVERAGE}"
+    VERSION               "${${PROJECT_NAME}_VERSION}"
```

Comment:
I believe `PROJECT_NAME` is set by the topmost `project(...)` command, so it would be safer to use `LIBRARY_NAME` here and in the following

Reply:
There is `PROJECT_NAME` and `CMAKE_PROJECT_NAME`. The former always refers to the innermost project, the latter always refers to the outermost project. That is, if somebody adds this library as subdirectory, CMAKE_PROJECT_NAME will refer to their project, and PROJECT_NAME will refer to this library.